samples/retained-instance-store/src/main/java/com/badoo/ribs/samples/retained_instance_store/rib/retainedInstance/routing/RetainedInstanceChildBuilders.kt

```diff
@@ -0,0 +1,25 @@
+package com.badoo.ribs.samples.retained_instance_store.rib.retainedInstance.routing
+
+import com.badoo.ribs.samples.retained_instance_store.rib.counter.Counter
+import com.badoo.ribs.samples.retained_instance_store.rib.counter.CounterBuilder
+import com.badoo.ribs.samples.retained_instance_store.utils.SecondsClock
+import com.badoo.ribs.samples.retained_instance_store.rib.retainedInstance.RetainedInstance
+
+
+internal open class RetainedInstanceChildBuilders(
+        private val dependency: RetainedInstance.Dependency
+) {
+    val retainedCounter: CounterBuilder
+        get() = getCounterChildBuilder(dependency, true)
+    val notRetainedCounter: CounterBuilder
+        get() = getCounterChildBuilder(dependency, false)
+
+    private fun getCounterChildBuilder(dependency: RetainedInstance.Dependency, isRetained: Boolean): CounterBuilder {
```

Comment:
You don't need this method if you treat `isRetained` as a build time dependency.

Reply:
No, the clock is fine as a dependency, I'm only talking about the isRetained flag.